### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by WhiteNight121**
i tired this step - Check out the "dev" branch: `git checkout -b dev` and get this error "fatal: A branch named 'dev' already exists" im thinking that step is not needed as after you clone it that branch is made for you yes no maybe?

### CODE REVIEW

File: CONTRIBUTING.md
Comment (py-crash) [inline CONTRIBUTING.md:90]: You shouldn't delete this line. You should change it to:

```markdown
Check out the "dev" branch: `git checkout dev`
```

Comment (WhiteNight121) [reply CONTRIBUTING.md:90]: when i did that i got "Already on 'dev' Your branch is up to date with 'origin/dev'." so if thats what we want to happen then we can make the change you suggested.

Comment (RealOrangeOne) [reply CONTRIBUTING.md:90]: `dev` is the default branch for the repo, so it's checked out by default.

**Review by RealOrangeOne — DISMISSED** [review]

File: CONTRIBUTING.md
Comment (py-crash) [reply CONTRIBUTING.md:90]: That's not really an error. It is better that you make sure that you are using `dev` and not some other branch as it could lead to errors

**Comment by Bas-Man** [comment]
This article has received one approval but no second approval. Thus it has not yet been merged in to dev.

Perhaps another editor could take a look. Check and check back with @RealOrangeOne and get this committed?

File: CONTRIBUTING.md
Comment (Bas-Man) [inline CONTRIBUTING.md:90]: - checkout a new branch with a descriptive name for your work. `git checkout -b ha-presence-detectiion-with-monitor` This is informative as it tells reviewers a little about what the branch will be committing.

File: CONTRIBUTING.md
Comment (Bas-Man) [inline CONTRIBUTING.md:92]: - Push your commit to your fork: ` git push origin ha-presence-detection-with-monitor`
- Generate a pull-request using GitHub's web interface.

File: CONTRIBUTING.md
Comment (Bas-Man) [inline CONTRIBUTING.md:89]: - Clone your fork to your local machine. and ....

**Review by Bas-Man — COMMENTED** [review]
Sorry, hope you guys don't mind me adding to this.
If I knew how I would have made the changes and pushed them to @WhiteNight121

**Comment by py-crash** [comment]
Try to communicate with @WhiteNight121 on Discord. If he is AWOL you can just fork his repo and create a new pull request with your changes from your own branch

**Comment by WhiteNight121** [comment]
We talked about I'll try and take a look at the comments soon.

**Comment by Bas-Man** [comment]
I have also made a pull request again WhiteNIght's Repo. I did that earlier.

File: CONTRIBUTING.md
Comment (WhiteNight121) [reply CONTRIBUTING.md:89]: they may not be on there local machine i think ill leave as is for now

File: CONTRIBUTING.md
Comment (WhiteNight121) [reply CONTRIBUTING.md:92]: im still learning git/github but im not talking about the github interface here unless im missing something

File: CONTRIBUTING.md
Comment (WhiteNight121) [reply CONTRIBUTING.md:90]: i wrote this some time ago now but if i remember this step is just checking out the branch to your local PC i did not think you could add descriptive names till you check it back in and add a comment? again i could be missing something

Comment (Bas-Man) [reply CONTRIBUTING.md:90]: After you clone the repo to your local machine. You would normally create a working branch. This is where you do your edits. It protects the dev branch. And allows you to pull in any upstream updates. This way you can hopefully avoid and fix merge conflicts. 

At least that is my usually work practice.

Comment (Bas-Man) [reply CONTRIBUTING.md:90]: For this particular repo. 'Dev' is the default branch. So there is no real need to do a checkout.

Comment (WhiteNight121) [reply CONTRIBUTING.md:90]: > 
> 
> After you clone the repo to your local machine. You would normally create a working branch. This is where you do your edits. It protects the dev branch. And allows you to pull in any upstream updates. This way you can hopefully avoid and fix merge conflicts.
> 
> At least that is my usually work practice.

understand but in this case are they not already working on there own repo?

Comment (WhiteNight121) [reply CONTRIBUTING.md:90]: > 
> 
> For this particular repo. 'Dev' is the default branch. So there is no real need to do a checkout.

ya i agree but thats how this repo is setup others may not be i was trying to make the steps more generic but of now its fine i can add back in if we get any more feed back on it for now im just going to approval all the changes. thanks!!

Comment (Bas-Man) [reply CONTRIBUTING.md:90]: They are working on their own repo. That is true.

But let's say you wanted to work on creating two seperate articles at the same time. I don think it would be possible to keep them as separate pull requests easily. You would need to finish both before they could be merged.
Having something like:
dev->new-branch-a
dev->new-branch-b
 
This would allow you to work on two articles at the same time. Without them effecting each other too much. The branch name also keep it clear what you are working on.

I think it makes it easier to keep you local 'dev' and GitHub 'dev' in sync with the upstream 'dev' located under self-hosted-show/wki

pull selfhosted/wiki/ -> WhiteNight121/wiki
pull origin(WhiteNIght121) -> local machine

If that makes sense.

Comment (Bas-Man) [reply CONTRIBUTING.md:90]: Keep in mind this just my subjective opinion. :)

Comment (WhiteNight121) [reply CONTRIBUTING.md:90]: sort of but also i dont think im at that level with git yet... but good to think about...

**Comment by IronicBadger** [comment]
@WhiteNight121 what's the status on this?

**Comment by WhiteNight121** [comment]
> 
> 
> @WhiteNight121 what's the status on this?

i merged @Bas-Man changes in last night so i think done....

**Comment by IronicBadger** [comment]
No longer needed? I'm going to close - ping me or reopen if you feel this is in error.

**Pull request closed without merging**
